src/material-experimental/mdc-progress-spinner/progress-spinner.ts

```diff
@@ -0,0 +1,242 @@
+   * 'determinate'.
+   * Mirrored to mode attribute.
+   */
+  @Input()
+  get mode(): ProgressSpinnerMode { return this._mode; }
+
+  set mode(value: ProgressSpinnerMode) {
+    this._mode = value;
+    this._syncFoundation();
+  }
+
+  private _value = 0;
+
+  /** Value of the progress bar. Defaults to zero. Mirrored to aria-valuenow. */
+  @Input()
+  get value(): number {
+    return this.mode === 'determinate' ? this._value : 0;
+  }
+
+  set value(v: number) {
+    this._value = Math.max(0, Math.min(100, coerceNumberProperty(v)));
+    this._syncFoundation();
+  }
+
+  private _diameter = BASE_SIZE;
+
+  /** The diameter of the progress spinner (will set width and height of svg). */
+  @Input()
+  get diameter(): number {
+    return this._diameter;
+  }
+
+  set diameter(size: number) {
+    this._diameter = coerceNumberProperty(size);
+    this._syncFoundation();
+  }
+
+  private _strokeWidth: number;
+
+  /** Stroke width of the progress spinner. */
+  @Input()
+  get strokeWidth(): number {
+    return this._strokeWidth ?? this.diameter / 10;
+  }
+
+  set strokeWidth(value: number) {
+    this._strokeWidth = coerceNumberProperty(value);
+  }
+
+  /** The radius of the spinner, adjusted for stroke width. */
+  _circleRadius(): number {
+    return (this.diameter - BASE_STROKE_WIDTH) / 2;
+  }
+
+  /** The view box of the spinner's svg element. */
+  _viewBox() {
+    const viewBox = this._circleRadius() * 2 + this.strokeWidth;
+    return `0 0 ${viewBox} ${viewBox}`;
+  }
+
+  /** The stroke circumference of the svg circle. */
+  _strokeCircumference(): number {
+    return 2 * Math.PI * this._circleRadius();
+  }
+
+  /** The dash offset of the svg circle. */
+  _strokeDashOffset() {
+    if (this.mode === 'determinate') {
+      return this._strokeCircumference() * (100 - this._value) / 100;
+    }
+    return null;
+  }
+
+  /** Stroke width of the circle in percent. */
+  _circleStrokeWidth() {
+    return this.strokeWidth / this.diameter * 100;
+  }
+
+  ngAfterViewInit() {
+    this._foundation = new MDCCircularProgressFoundation(this._adapter);
+    this._foundation.init();
+    this._syncFoundation();
+  }
+
+  ngOnDestroy() {
+    if (this._foundation) {
+      this._foundation.destroy();
+    }
+  }
+
+  /** Syncs the state of the progress spinner with the MDC foundation. */
+  private _syncFoundation() {
+    const foundation = this._foundation;
+
+    if (foundation) {
+      const mode = this.mode;
+      foundation.setProgress(this.value / 100);
+      foundation.setDeterminate(mode === 'determinate');
+    }
+  }
+
+  static ngAcceptInputType_diameter: NumberInput;
+  static ngAcceptInputType_strokeWidth: NumberInput;
+  static ngAcceptInputType_value: NumberInput;
+}
+
+/**
+ * `<mat-spinner>` component.
+ *
+ * This is a component definition to be used as a convenience reference to create an
+ * indeterminate `<mat-progress-spinner>` instance.
+ */
+export {MatProgressSpinner as MatSpinner}
```

Comment:
Btw, bonus points if you want to make this same alias change for the existing mat-spinner in another PR. I think it will cut down on code size a bit.

Reply:
https://github.com/angular/components/pull/14514